app/lib/common/actions/show_limited_space_list.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/widgets/spaces/space_card.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+Future<void> showLimitedSpaceList(
+  BuildContext context,
+  WidgetRef ref,
+  String roomId,
+) async {
+  final limitedSpaceIds =
+      await ref.watch(joinRulesAllowedRoomsProvider(roomId).future);
+  if (!context.mounted) return;
+  showModalBottomSheet(
+    showDragHandle: true,
+    context: context,
+    constraints: const BoxConstraints(maxHeight: 450),
+    isScrollControlled: true,
+    builder: (context) {
+      return _buildLimitedSpaceListUI(context, ref, limitedSpaceIds);
+    },
+  );
+}
+
+Widget _buildLimitedSpaceListUI(
```

Comment:
May I suggest to put this into `ConsumerWidget` (with a `this.roomId`), then the `ref.watch` could be moved here and it would actually update properly on changes, too..

Also, it is a bit nicer ;) .

Reply:
Sure